keras_cv/visualization/plot_gallery.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+import keras_cv
+from keras_cv.utils import assert_matplotlib_installed
+
+try:
+    import matplotlib.pyplot as plt
+except:
+    plt = None
+
+
+def plot_gallery(
```

Comment:
I think I'd expect `plot_gallery` to support both the with & without boxes use cases, based on whether there is a `"bounding_boxes"` key in the input data.

Is there a technical limitation that prevents this?

Reply:
That would be a much better UX - thanks for pointing this out.

Reply:
+1
super-nice if this supports the exact same dict format as models and KPLs.

Reply:
Alright, so I added `plot_image_gallery()` instead of `plot_gallery()`.

I've done a lot of thoughtwork and prototyping and I'm convinced that the signature for a mega utility that plots everything is too ugly when you expose all of the parameters...

What we should do is offer:

```
visualization.visualize_dataset()
```

To tackle this use case instead.